x-pack/plugins/security_solution/public/common/components/matrix_histogram/index.tsx

```diff
@@ -178,93 +104,46 @@ export const MatrixHistogramComponent: React.FC<MatrixHistogramComponentProps> =
   );
 
   const { toggleStatus, setToggleStatus } = useQueryToggle(id);
-  const [querySkip, setQuerySkip] = useState(skip || !toggleStatus);
-  useEffect(() => {
-    setQuerySkip(skip || !toggleStatus);
-  }, [skip, toggleStatus]);
+
   const toggleQuery = useCallback(
     (status: boolean) => {
       setToggleStatus(status);
-      // toggle on = skipQuery false
-      setQuerySkip(!status);
     },
-    [setQuerySkip, setToggleStatus]
+    [setToggleStatus]
   );
 
-  const isChartEmbeddablesEnabled = useIsExperimentalFeatureEnabled('chartEmbeddablesEnabled');
-
-  const matrixHistogramRequest = {
-    endDate,
-    errorMessage,
-    filterQuery,
-    histogramType,
-    indexNames,
-    onError,
-    startDate,
-    stackByField: selectedStackByOption.value,
-    runtimeMappings,
-    isPtrIncluded,
-    skip: querySkip || isChartEmbeddablesEnabled,
-  };
-  const [loading, { data, inspect, totalCount, refetch }] =
-    useMatrixHistogramCombined(matrixHistogramRequest);
-
   const titleWithStackByField = useMemo(
     () => (title != null && typeof title === 'function' ? title(selectedStackByOption) : title),
     [title, selectedStackByOption]
   );
-  const { responses } = useVisualizationResponse({ visualizationId });
+  const { responses: visualizationResponse } = useVisualizationResponse({ visualizationId });
```

Comment:
This is still an array, consider keeping the plural in the name:
```suggestion
  const { responses: visualizationResponses } = useVisualizationResponse({ visualizationId });
```